src/value.rs

```diff
@@ -665,3 +665,312 @@ impl Value {
+    /// #
+    /// # fn main() {
+    /// use serde_yaml::Value;
+    ///
+    /// let v = vec!["lorem", "ipsum", "dolor"];
+    /// let x: Value = v.into();
+    /// # }
+    /// ```
+    fn from(f: Vec<T>) -> Self {
+        Value::Sequence(f.into_iter().map(Into::into).collect())
+    }
+}
+
+impl<'a, T: Clone + Into<Value>> From<&'a [T]> for Value {
+    /// Convert a slice to `Value`
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # extern crate serde_yaml;
+    /// #
+    /// # fn main() {
+    /// use serde_yaml::Value;
+    ///
+    /// let v: &[&str] = &["lorem", "ipsum", "dolor"];
+    /// let x: Value = v.into();
+    /// # }
+    /// ```
+    fn from(f: &'a [T]) -> Self {
+        Value::Sequence(f.into_iter().cloned().map(Into::into).collect())
+    }
+}
+
+use std::iter::FromIterator;
+
+impl<T: Into<Value>> FromIterator<T> for Value {
+    /// Convert an iteratable type to a YAML sequence
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # extern crate serde_yaml;
+    /// #
+    /// # fn main() {
+    /// use serde_yaml::Value;
+    ///
+    /// let v = std::iter::repeat(42).take(5);
+    /// let x: Value = v.collect();
+    /// # }
+    /// ```
+    ///
+    /// ```rust
+    /// # extern crate serde_yaml;
+    /// #
+    /// # fn main() {
+    /// use serde_yaml::Value;
+    ///
+    /// let v: Vec<_> = vec!["lorem", "ipsum", "dolor"];
+    /// let x: Value = v.into_iter().collect();
+    /// # }
+    /// ```
+    ///
+    /// ```rust
+    /// # extern crate serde_yaml;
+    /// #
+    /// # fn main() {
+    /// use std::iter::FromIterator;
+    /// use serde_yaml::Value;
+    ///
+    /// let x: Value = Value::from_iter(vec!["lorem", "ipsum", "dolor"]);
+    /// # }
+    /// ```
+    fn from_iter<I: IntoIterator<Item = T>>(iter: I) -> Self {
+        let vec: Vec<Value> = iter.into_iter().map(|x| x.into()).collect();
+
+        Value::Sequence(vec)
+    }
+}
+
+impl<T: Into<Value>, U: Into<Value>> FromIterator<(T, U)> for Value {
+    /// Convert an iteratable type to a YAML mapping
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # extern crate serde_yaml;
+    /// #
+    /// # fn main() {
+    /// use serde_yaml::Value;
+    ///
+    /// let x: Value = vec![("a", 42), ("b", 21)].into_iter().collect();
+    /// # }
+    /// ```
+    ///
+    /// ```rust
+    /// # extern crate serde_yaml;
+    /// #
+    /// # fn main() {
+    /// use serde_yaml::Value;
+    ///
+    /// let v: Vec<_> = vec![("lorem", "ipsum"), ("dolor", "sit amet")];
+    /// let x: Value = v.into_iter().collect();
+    /// # }
+    /// ```
+    ///
+    /// ```rust
+    /// # extern crate serde_yaml;
+    /// #
+    /// # fn main() {
+    /// use std::iter::FromIterator;
+    /// use serde_yaml::Value;
+    ///
+    /// let x: Value = Value::from_iter(vec![("lorem", "ipsum"), ("dolor", "sit amet")]);
+    /// # }
+    /// ```
+    fn from_iter<I: IntoIterator<Item = (T, U)>>(iter: I) -> Self {
```

Comment:
This whole thing is sadly a bit fragile, as `Iterator<&(_, _)>` gets caught by the sequence impl `impl<T: Into<Value>> FromIterator<T> for Value` :/